### PR TITLE
Add WT3 crystallizer as a subtype of brine concentrator

### PR DESCRIPTION
**Pull request by luohezhiming**
## Summary/Motivation:

Add WT3 crystallizer unit model as a subtype model of brine concentrator unit model

## Changes proposed in this PR:
- add crystallizer as a subtype in the brine_concentrator yaml file
- add electric coefficients as varaibles
- fixed a typo in the original brince_concentrator_zo.py

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Comment by luohezhiming** [comment]
@adam-a-a It seems only the check for nmacOS setup failed. Do you have any idea what happened here?

**Comment by bknueven** [comment]
> @adam-a-a It seems only the check for nmacOS setup failed. Do you have any idea what happened here?

I think #393 will fix this, as this failing test will be skipped. The conda-ipopt was updated earlier today, which I expect is what's causing this sensitive test to fail.

**Comment by luohezhiming** [comment]
> I think https://github.com/watertap-org/watertap/pull/393 will fix this, as this failing test will be skipped. The conda-ipopt was updated earlier today, which I expect is what's causing this sensitive test to fail.

OK. Will rerun it when #393 get merged.

File: watertap/data/techno_economic/brine_concentrator.yaml [hunk 1]
```diff
@@ -144,3 +160,166 @@ default:
...
+  elec_coeff_3:
+    value: 9.47
+    units: kWh/m^3
+    reference: Survey of High-Recovery and Zero Liquid Discharge Technologies for Water Utilities (2008).
+  elec_coeff_4:
+    value: 8.63E-4
+    units: kWh/m^6*hr
+    reference: Survey of High-Recovery and Zero Liquid Discharge Technologies for Water Utilities (2008).
+  recovery_frac_mass_H2O:
+    value: 0.95
```
Comment (kurbansitterley) [inline watertap/data/techno_economic/brine_concentrator.yaml:182]: @luohezhiming did you change this value? Nothing wrong with it if there is a reference, I just know water_recovery is 0.9 in WT3.

Comment (luohezhiming) [reply watertap/data/techno_economic/brine_concentrator.yaml:182]: I copy this water_recovery from the original crystallizer.yaml file. The water_recovery for a brine_concentrator is 0.9, but is given 0.95 in the original crystallizer.yaml file.

**Review by adam-a-a — APPROVED** [review]
LGTM!

**Review by bknueven — COMMENTED** [review]
Seems reasonable -- should we be explicitly testing the new subtype?

File: watertap/data/techno_economic/brine_concentrator.yaml (hunk 1 above)
Comment (kurbansitterley) [reply watertap/data/techno_economic/brine_concentrator.yaml:182]: Ah perfect, I missed that this was for the subtype

**Comment by kurbansitterley** [comment]
I am not a pro on what warrants a separate test, but my gut says yes since they recovery is different, they would have different flows

**Comment by luohezhiming** [comment]
@bknueven @kurbansitterley The original test file include a subtype testing `class Testbrine_concentratorZOsubtype`, I think it is OK because other units we migrated that has subtypes are also tested by this way.

**Review by andrewlee94 — APPROVED** [review]
My only comment is that I don't think we want the electricity intensity coefficients in the report, but that can be cleaned up later.

**Pull request merged**